GooglePlayInstant/Editor/PlayInstantLoadingScreenGenerator.cs

```diff
@@ -91,29 +104,19 @@ private static void AddLoadingScreenScript(GameObject loadingScreenGameObject)
             loadingScreenImage.sprite = loadingImageSprite;
         }
 
-        //TODO: add better handling of finding assets folder and figure out possible alternative AssetDatabase synchronous importing
-        private static void GenerateLoadingScreenScript(string assetBundleUrl)
+        private static void GenerateLoadingScreenConfigfile()
         {
-            var newLoadingScreenScriptPath = "Assets/GooglePlayInstantScript/LoadingScreenScript.cs";
-            var genericLoadingScriptDirStrings = Directory.GetFiles(Directory.GetCurrentDirectory(),
-                "GenericLoadingScreenScript.cs",
-                SearchOption.AllDirectories);
-            if (genericLoadingScriptDirStrings.Length == 0)
-            {
-                Debug.LogErrorFormat("Generic Loading Script could not be found in current project directory: {0}",
-                    Directory.GetCurrentDirectory());
-            }
-            else
-            {
-                var genericLoadingScreenScriptDir = genericLoadingScriptDirStrings[0];
-                Directory.CreateDirectory(Directory.GetParent(newLoadingScreenScriptPath).FullName);
-
-                var genericLoadingScreenScript = File.ReadAllText(genericLoadingScreenScriptDir);
-                var newLoadingScreenScript = genericLoadingScreenScript.Replace("__ASSETBUNDLEURL__", assetBundleUrl)
-                    .Replace("GenericLoadingScreenScript", "LoadingScreenScript");
-                File.WriteAllText(newLoadingScreenScriptPath, newLoadingScreenScript);
-                AssetDatabase.Refresh(ImportAssetOptions.ForceSynchronousImport);
-            }
+            var newLoadingScreenConfigObj =
+                new LoadingScreenConfig {assetBundleUrl = QuickDeployConfig.Config.assetBundleUrl};
+
+            var newLoadingScreenConfigJson = EditorJsonUtility.ToJson(newLoadingScreenConfigObj);
+
+            File.WriteAllText(Path.Combine(LoadingScreenJsonPath, LoadingScreenJsonFileName),
+                newLoadingScreenConfigJson);
+
+            // Force asset to import synchronously so that testing can be completed immediately after generating a loading screen.
+            AssetDatabase.ImportAsset(Path.Combine(LoadingScreenJsonPath, LoadingScreenJsonFileName),
```

Comment:
The expression Path.Combine(LoadingScreenJsonPath, LoadingScreenJsonFileName) is also on line 114. This should be extracted as a local variable (or a field) and reused.